File: problem-2/2.py
import math

max = 4000000
sum = 0

def fibonacci(n): 
    phi = (1 + math.sqrt(5)) / 2
    return round(pow(phi, n) / math.sqrt(5))

def findEvenFib(n): 
    global sum
    if fibonacci(n) >= max: return sum
    else:
        sum += fibonacci(n) if not (fibonacci(n) & 1) else 0
        return findEvenFib(n + 1)

print(findEvenFib(sum))

File: problem-2/2.cpp
#include <bits/stdc++.h>
using namespace std;

int MAX = 4000000;
int sum = 0;

int fibonacci(int n) {
    double phi = (1 + sqrt(5)) / 2; 
    return round(pow(phi, n) / sqrt(5)); 
}

void findEvenFib(int n) {
    if(fibonacci(n) >= MAX) { return; }
    sum += (!(fibonacci(n) & 1)) ? fibonacci(n) : 0;
    findEvenFib(n + 1); 
}

int main(void) {
    findEvenFib(sum);
    cout << sum << endl;
    return 0;
}